projects/observability/src/shared/components/gauge/gauge.component.ts

```diff
@@ -117,13 +117,32 @@ export class GaugeComponent implements OnChanges {
   }
 
   private buildRadius(boundingBox: ClientRect): number {
-    return Math.min(boundingBox.height - GaugeComponent.GAUGE_AXIS_PADDING, boundingBox.width / 2);
+    const width = boundingBox.width;
+    const height = boundingBox.height;
+
+    if (height > width / 2) {
+      /**
+       * Since height > width > 2, radius can be treated as (width / 2), but only for semi-circles
+       *
+       * If there is an extra angle added to the semicircle,
+       * radius + extra arc length can over shoots the height
+       *
+       * If radius + extra arc length over shoots the height, then radius
+       * needs to be calculated using the provided height, instead of width / 2
+       */
+      const radius = width / 2;
+      const extraArcHeight = this.calculateExtraArcLength(radius);
```

Comment:
If we adjust the text position, could we avoid all this additional logic due to the extra arc length?